### PR TITLE
Handle symbolic links as source files

### PR DESCRIPTION
**Pull request by ralf1070**
When replaying content via DLNA 'Microsoft.AspNetCore.Http.SendFileResponseExtensions.SendFileAsyncCore'
validates the length of the sent information. Unfortuanately this check fails for source files
which are symbolic links (aka ReparsePoint's). In this case the size of the filename of the referenced
file is reported, instead of the size of the content of this file.

  * To fix this in AspNetCore.Http would be best, but is out of scope.
  * Second best would be, to resolve the symbolic link target and return this as PhysicalFileResult. This
    would require to add Mono.Posix to this project, which would add another Assembly and which is - as far
    as I know - not portable to Windows and therefore would require conditional compilation.
  * The third way is to use FileStreamResult for files known to be symbolic links, as it correctly reports
    the size of the open file. For all other files PhysicalFileResult is kept, as other parts of this
    project may rely on using it's properties.

### CODE REVIEW

**Comment by genofire** [comment]
I use as file backend, an `git annex` repo - i really need this - any update?

File: Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs [hunk 1]
```diff
@@ -75,6 +75,27 @@ public static class FileStreamResponseHelpers
...
+            //
+            // * To fix this in AspNetCore.Http would be best, but is out of scope.
+            // * Second best would be, to resolve the symbolic link target and return this as PhysicalFileResult. This
+            //   would require to add Mono.Posix to this project, which would add another Assembly and which is - as far
+            //   as I know - not portable to Windows and therefore would require conditional compilation.
+            // * The third way is to use FileStreamResult for files known to be symbolic links, as it correctly reports
+            //   the size of the open file. For all other files PhysicalFileResult is kept, as other parts of this
+            //   project may rely on using it's properties.
+            var fileInfo = System.IO.File.GetAttributes(path);
+            if (fileInfo.HasFlag(FileAttributes.ReparsePoint))
```
Comment (cvium) [inline Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs:93]: HasFlag isn't very efficient
```suggestion
            if ((fileInfo.Attributes & FileAttributes.ReparsePoint) == FileAttributes.ReparsePoint)
```

**Comment by Bond-009** [comment]
As mentioned in #5775 the ideal solution is a combination of the 2. The no dep solution of this with the drop in usability of the other one.

**Comment by Bond-009** [comment]
Superseded by #6541

**Pull request closed without merging**